Resview gallery

vlukes opened the pull request:
This PR modifies `script/gen_gallery.py` in order to use `resview.py` instead of using `sfepy.postprocessing` for generating the image gallery.

rc commented on resview.py:
```diff
@@ -702,17 +754,17 @@ def main():
-                    nm.max([iv[0][1] for iv in v]))
-            scalar_bars[k] = (clim, v[0][1])
+        for k in scalar_bar_limits.keys():
+            lims = scalar_bars[k]
```
Pyflakes says that `scalar_bars` is undefined here.

vlukes replied:
Of course, Pyflakes is right, there should be `scalar_bars_limits`.

vlukes replied:
I forgot to test the animations.

rc commented:
Thanks, nice! Just a few notes:

- I got encoding errors when writing some DG examples:
  - In examples/dg/advection_diffusion_2D.py replace, please, the long dash in `convection-diffusion` in the docstring by the regular dash.
  - In examples/dg/burgers_2D.py, there is `Kučera`, which is correct, so use
    ```
    fd = open(full_rst_filename, 'w', encoding='utf-8')
    ```
    in `generate_rst_files()`.
- For 2D flow-related problems, it may be nice to show the streamlines with or without the glyphs, depending on the mesh size (Can pyvista decimate the glyphs?)
- The piezo-elasticity figure has 6 plots but only 4 scalar bars (and phi is not plotted, or is it outside the view?).
- 2D figures could be more zoomed in to reduce the border size.
- In 3D figures with many plots the border plots are clipped.
- Generally the figures with many plots could use a grid layout, as in the old gallery. Maybe next PR? See the `--layout` option of `postproc.py`.

For future PR:
- Similar to streamlines, other kinds of plots could be useful, e.g. iso-surfaces (see `--3d`, `--scalar-mode`, `--vector-mode` options of `postproc.py`. Other options might be also interesting, e.g. ``, but maybe it could go to `show_mesh_info.py` or a new script. I am writing it here not to forget.

rc commented:
Other notes:

- The example docstrings mostly show how to use `postproc.py`, but  the images are created by `resview.py` now, so the description does not correspond to what is shown. The docstrings need to be updated. (Another PR :)
- Some view angles/glyph sizes etc. could use some more tweaking (e.g. the gradient in `diffusion/sinbc.py`). It could be done possibly in connection with the above point...

But overall - great job!

vlukes commented:
> Thanks, nice! Just a few notes:
> 
>     * I got encoding errors when writing some DG examples:
>       
>       * In examples/dg/advection_diffusion_2D.py replace, please, the long dash in `convection-diffusion` in the docstring by the regular dash.
>       * In examples/dg/burgers_2D.py, there is `Kučera`, which is correct, so use
>         ```
>         fd = open(full_rst_filename, 'w', encoding='utf-8')
>         ```  
>         in `generate_rst_files()`.

Fixed

>     * For 2D flow-related problems, it may be nice to show the streamlines with or without the glyphs,
>       depending on the mesh size (Can pyvista decimate the glyphs?)

I will add a new enhancement issue related to this (#757).

>     * The piezo-elasticity figure has 6 plots but only 4 scalar bars (and phi is not plotted,
>       or is it outside the view?).

The other plots are outside the view. Now, the number of plots  is reduced to 4.

>     * 2D figures could be more zoomed in to reduce the border size.

2D plots are produced using `view_xy()` function. But actually I do not know, how to change the zoom factor for this.

>     * In 3D figures with many plots the border plots are clipped.

Fixed.

>     * Generally the figures with many plots could use a grid layout, as in the old gallery. Maybe next PR?
>       See the `--layout` option of `postproc.py`.

Good idea! See enhancement issue #758.

> 
> For future PR:
> 
>     * Similar to streamlines, other kinds of plots could be useful, e.g. iso-surfaces (see `--3d`, `--scalar-mode`, `--vector-mode`
>       options of `postproc.py`. Other options might be also interesting, e.g. ``, but maybe it could go to `show_mesh_info.py`
>       or a new script. I am writing it here not to forget.

OK, #759 created.

vlukes commented:
>

vlukes commented:
> Other notes:
> 
>     * The example docstrings mostly show how to use `postproc.py`, but  the images are created by `resview.py` now, so the description does not correspond to what is shown. The docstrings need to be updated. (Another PR :)
> 
>     * Some view angles/glyph sizes etc. could use some more tweaking (e.g. the gradient in `diffusion/sinbc.py`). It could be done possibly in connection with the above point...

New issue #760.

rc commented:
Thanks let's us merge this and work on the subsequent PRs. LGTM!
